File: problemas/iniciante/1160/1160.c
#include <stdio.h>
#include <math.h>

int main()
{
    int T, PA, PB;
    double G1, G2;

    scanf("%d", &T);
    for (int i = 0; i < T; ++i)
    {
        scanf("%d %d %lf %lf", &PA, &PB, &G1, &G2);
        G1 /= 100;
        G2 /= 100;

        int anos;
        for (anos = 1; PA <= PB && anos < 101; ++anos)
        {
            PA += floor(PA * G1);
            PB += floor(PB * G2);
        }

        if (PA <= PB)
        {
            printf("Mais de 1 seculo.\n");
        }
        else
        {
            printf("%d anos.\n", anos - 1);
        }
    }

    return 0;
}

File: problemas/iniciante/1160/1160.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.lang.Math;

public class Main {
    public static void main(String[] args) throws IOException {
        InputStreamReader ir = new InputStreamReader(System.in);
        BufferedReader in = new BufferedReader(ir);

        int T = Integer.parseInt(in.readLine());
        for (int i = 0; i < T; ++i) {
            String[] entrada = in.readLine().trim().split(" ");
            int PA = Integer.parseInt(entrada[0]);
            int PB = Integer.parseInt(entrada[1]);
            double G1 = Double.parseDouble(entrada[2]) / 100;
            double G2 = Double.parseDouble(entrada[3]) / 100;

            int anos;
            for (anos = 1; PA <= PB && anos < 101; ++anos) {
                PA += Math.floor(PA * G1);
                PB += Math.floor(PB * G2);
            }

            if (PA <= PB) {
                System.out.println("Mais de 1 seculo.");
            } else {
                System.out.printf("%d anos.\n", anos - 1);
            }
        }
    }
}